app/models/user.rb

```diff
@@ -3,4 +3,15 @@
 class User < ApplicationRecord
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
+  has_one_attached :avatar
+  validate :avatar_type
+
+  private
+
+  def avatar_type
+    return unless avatar.attached? && !avatar.blob.content_type.in?(%('image/jpg image/png image/gif'))
+
+    avatar.purge
+    errors.add(User.human_attribute_name(:avatar), 'はjpgまたはpngまたはgif形式でアップロードしてください')
```

Comment:
```suggestion
    errors.add(:avatar, 'はjpgまたはpngまたはgif形式でアップロードしてください')
```

Railsの仕組み的にこれでよかったりしませんか？

Reply:
ありがとうございます。
ご提案いただいたとおり変更し、`:avatar`が`アイコン`と翻訳されることを確認しました。
今回、`:avatar`だけで`アイコン`と翻訳されたのは、以下のような理由からと認識していますが合っていますでしょうか？

- `translate_ja.yml`ファイルの中で、`User`モデルの中に`avatar: アイコン`と翻訳を設定している
- 今回は、`User`モデルを定義する`app/models/user.rb`の中で`:avatar`が呼び出されたため、`:avatar`のみで「avatar: アイコン」という翻訳が適用された